catroid/src/test/java/org/catrobat/catroid/test/content/actions/WebRequestActionTest.java

```diff
@@ -185,4 +190,50 @@ public void testCancelledCallAndResendRequest() {
 		verify(StageActivity.stageListener.webConnectionHolder, times(2)).addConnection(any());
 		verify(webConnection, times(2)).sendWebRequest();
 	}
+
+	@Test
+	public void testSuccessfulResponseWithInputVariable() {
+		UserVariable inputVariable = new UserVariable(TEST_INPUTVARIABLE, TEST_URL);
+		FormulaElement formulaElement = new FormulaElement(
+				FormulaElement.ElementType.USER_VARIABLE, inputVariable.getName(), null);
+		final String responseString = "Response";
+
+		Formula formula = Mockito.spy(new Formula(formulaElement));
+
+		testSprite.addUserVariable(userVariable);
+		testSprite.addUserVariable(inputVariable);
+
+		WebRequestAction action = (WebRequestAction) testSprite.getActionFactory().createWebRequestAction(
+				testSprite,
+				formula,
+				userVariable
+		);
+		action.setWebConnectionFactory(webConnectionFactory);
+
+		when(StageActivity.stageListener.webConnectionHolder.addConnection(any())).thenReturn(true);
+
+		try {
+			doAnswer(invocation -> {
+				Sprite sprite = invocation.getArgument(0);
+				return String.valueOf(sprite.getUserVariable(TEST_INPUTVARIABLE).getValue());
+			}).when(formula).interpretString(any(Sprite.class));
+		} catch (InterpretationException e) {
+			fail("Formula could not be interpreted");
```

Comment:
An Exception will automatically cause the test to fail. So please dont catch Excpetions just to make the test "fail". Also think about another test where this failing could be triggered in a way that you want the test to pass if the Exception is thrown. Hint: Annotations `@Test(expected = NullPointerException.class)` like this will make the test pass if the mentioned exception is thrown.

Reply:
I removed the try-catch-block, and added the exception to the method signature.

I choose to not add a test for testing the failing exception, because the exception is triggered by the function **Formula.interpretString**, which is not to test in this test-class, and the function is mocked anyway.
Please let me know, if I should add a test for it.